client/src/global/components/LanguageSelect/index.js

```diff
@@ -36,7 +36,8 @@ function LanguageSelect({ authentication, language }) {
   useEffect(() => {
     setLang(initialLanguage);
     updateI18n(initialLanguage);
-  }, [initialLanguage]);
+    updateLanguage(initialLanguage);
+  }, [initialLanguage, updateLanguage]);
 
   return (
     <Select
```

Comment:
@zdavis `state.ui.persistent.locale.language` was resetting to "en" on every route change. I assumed this wasn't the expected behavior and updated this `useEffect` to fix it. Was that the right thing to do?

Reply:
That doesn't seem right. I think this component may need another pass to get the store, `i18next`, and the internal state working in sync.